resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/security/HadoopFSCredentialProvider.scala

```diff
@@ -56,14 +57,14 @@ private[security] class HDFSCredentialProvider extends ServiceCredentialProvider
     }
 
     // Get the time of next renewal.
-    tokenRenewalInterval.map { interval =>
-      creds.getAllTokens.asScala
-        .filter(_.getKind == DelegationTokenIdentifier.HDFS_DELEGATION_KIND)
+    tokenRenewalInterval.flatMap { interval =>
+      val nextRenewalDates = creds.getAllTokens.asScala
+        .filter(_.decodeIdentifier().isInstanceOf[AbstractDelegationTokenIdentifier])
```

Comment:
Can we make sure all instances of `AbstractDelegationTokenIdentifier` are tokens for Hadoop file systems? Isn't Hive's `DelegationTokenIdentifier` also an `AbstractDelegationTokenIdentifier`?